lib/ohai/plugins/freebsd/cpu.rb

```diff
@@ -54,13 +57,14 @@
         # Features2=0x80000001<SSE3,<b31>>
       when /Features2=[a-f\dx]+<(.+)>/
         cpuinfo["flags"].concat($1.downcase.split(','))
-      when /Logical CPUs per core: (\d+)/
-        cpuinfo["cores"] = $1
+      when /FreeBSD\/SMP: Multiprocessor System Detected: (\d*) CPUs/
+        cpuinfo["total"] = $1.to_i
+      when /FreeBSD\/SMP: (\d*) package\(s\) x (\d*) core\(s\)/
+        cpuinfo["real"] = $1.to_i
+        cpuinfo["cores"] = $2.to_i
```

Comment:
No, I'm talking about the 'cores' computation.  If this is a dual-socket quad-proc shouldn't that look like:

```
FreeBSD/SMP: Multiprocessor System Detected: 16 CPUs
FreeBSD/SMP: 2 package(s) x 4 core(s) x 2 SMT threads
```

and cpuinfo["cores"] should be 8 (2 x 4)